controllers/hcpvaultsecretsapp_controller.go

```diff
@@ -123,6 +129,39 @@ func (r *HCPVaultSecretsAppReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, r.handleDeletion(ctx, o)
 	}
 
+	// keep track of the HCPVaultSecretsApp for the cleanup goroutine
+	if HCPVaultSecretsAppMap[req.NamespacedName.String()] == nil {
+		HCPVaultSecretsAppMap[req.NamespacedName.String()] = o
+	}
+
+	if !initializedOrphanedShadowSecretCleanup {
```

Comment:
I think that this was the quickest way to prevent a new goroutine coming up every time a CRUD operation is ran on an HVS app. We only need one because the goroutine iterates through every HVS app that had Reconcile() executed on it to check for a possibility of orphaned secrets.